### DISCUSSION TITLE
Connection error

### DISCUSSION THREAD

**Discussion by Timpearc** [ultralytics/ultralytics#7387] [Q&A]
Hi all,
I’m just getting started trying to train yolo with a custom dataset. I get an error 
Connection error download failure https//github.com/ultralytics/assets/releases/download/c0.0.0/yolov8n.pt environment is not online. 
this is run from a HPC virtual environment where only certain packages have internet accces 
Please advise how to proceed 
Many thanks 
Tim

**Comment by pderrenger** [comment]
@Timpearc hi Tim 👋,

It seems like your HPC environment has restricted internet access, which is preventing the automatic download of the `yolov8n.pt` pretrained weights. To proceed, you'll need to manually download the weights file on a machine with internet access and then transfer it to your HPC environment.

Here's what you can do:

1. On a machine with internet access, download the `yolov8n.pt` file from the Ultralytics release assets.
2. Transfer the downloaded `.pt` file to your HPC environment using a method that's allowed, such as `scp`, `rsync`, or any other file transfer method your HPC supports.
3. Once the file is on your HPC, you can point to it directly when creating the YOLO model instance in your training script.

For example, if you've placed `yolov8n.pt` in a directory named `/path/to/weights/`, you would modify your script to load the model like this:

```python
from ultralytics import YOLO

# Load the pretrained weights from the file you transferred
model = YOLO('/path/to/weights/yolov8n.pt')

# Continue with your training setup...
```

Make sure to replace `/path/to/weights/` with the actual path to the directory where you stored the weights file on your HPC system.

This should allow you to bypass the need for an internet connection to download the weights within your script. Good luck with your training! 🚀

If you need further guidance on training, you can refer to our [Train documentation](https://docs.ultralytics.com/modes/train).

**Reply by Timpearc** [reply]
That’s fab thanks I will give it a try

**Reply by glenn-jocher** [reply]
@Timpearc you're welcome, Tim! If you encounter any more issues or have questions along the way, don't hesitate to reach out. Happy training and best of luck with your custom dataset! 😊👍

**Reply by Timpearc** [reply]
hi think i'm having a silly moment the following link dosent have a yolov8n.pt weights file? ive poked around and cant see it in a different location. i installed using pip install ultralytics if that makes a difference
[https://github.com/ultralytics/ultralytics/tree/main/ultralytics/assets](url)
kind regards
tim

**Reply by glenn-jocher** [reply]
@Timpearc assets are hosted at https://github.com/ultralytics/assets/releases/tag/v8.1.0
